extensions/wikia/ArticleComments/ArticleComments_setup.php

```diff
@@ -11,48 +11,51 @@
+ * @license https://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
  * @package MediaWiki
  *
  * To activate this functionality, place this file in your extensions/
  * subdirectory, and add the following line to LocalSettings.php:
- *     require_once("$IP/extensions/wikia/ArticleComments/ArticleComments_setup.php");
+ *     require_once('$IP/extensions/wikia/ArticleComments/ArticleComments_setup.php');
  */
 
 
-$wgExtensionCredits['other'][] = array(
+$wgExtensionCredits['other'][] = [
 	'name' => 'ArticleComments',
 	'version' => '2.0',
-	'author' => array('[http://www.wikia.com/wiki/User:Eloy.wikia Krzysztof Krzyżaniak (eloy)]', '[http://www.wikia.com/wiki/User:Marooned Maciej Błaszkowski (Marooned)]'),
+	'author' => [ '[http://www.wikia.com/wiki/User:Eloy.wikia Krzysztof Krzyżaniak (eloy)]', '[http://www.wikia.com/wiki/User:Marooned Maciej Błaszkowski (Marooned)]' ],
 	'url' => 'https://github.com/Wikia/app/tree/dev/extensions/wikia/ArticleComments',
 	'descriptionmsg' => 'article-comments-desc'
-);
+];
 
-define('ARTICLECOMMENTORDERCOOKIE_NAME', 'articlecommentorder');
-define('ARTICLECOMMENTORDERCOOKIE_EXPIRE', 60 * 60 * 24 * 365);
-define('ARTICLECOMMENT_PREFIX', '@comment-');
-
-$dir = dirname(__FILE__);
+define( 'ARTICLECOMMENTORDERCOOKIE_NAME', 'articlecommentorder' );
+define( 'ARTICLECOMMENTORDERCOOKIE_EXPIRE', 60 * 60 * 24 * 365 );
+define( 'ARTICLECOMMENT_PREFIX', '@comment-' );
 
 // autoloaded classes
-$wgAutoloadClasses['ArticleCommentInit'] = "$dir/classes/ArticleCommentInit.class.php";
-$wgAutoloadClasses['ArticleComment'] = "$dir/classes/ArticleComment.class.php";
-$wgAutoloadClasses['ArticleCommentList'] = "$dir/classes/ArticleCommentList.class.php";
-$wgAutoloadClasses['ArticleCommentsAjax'] = "$dir/classes/ArticleCommentsAjax.class.php";
-$wgAutoloadClasses['ArticleCommentsController'] = "$dir/modules/ArticleCommentsController.class.php";
-$wgAutoloadClasses['CommentsIndex'] = "$dir/classes/CommentsIndex.class.php";
+$wgAutoloadClasses['ArticleCommentInit'] = __DIR__ . '/classes/ArticleCommentInit.class.php';
+$wgAutoloadClasses['ArticleComment'] = __DIR__ . '/classes/ArticleComment.class.php';
+$wgAutoloadClasses['ArticleCommentList'] = __DIR__ . '/classes/ArticleCommentList.class.php';
+$wgAutoloadClasses['ArticleCommentsAjax'] = __DIR__ . '/classes/ArticleCommentsAjax.class.php';
+$wgAutoloadClasses['ArticleCommentsController'] = __DIR__ . '/modules/ArticleCommentsController.class.php';
+$wgAutoloadClasses['CommentsIndex'] = __DIR__ . '/classes/CommentsIndex.class.php';
 
-$wgExtensionMessagesFiles['ArticleComments'] = dirname(__FILE__) . '/ArticleComments.i18n.php';
+$wgExtensionMessagesFiles['ArticleComments'] = __DIR__ . '/ArticleComments.i18n.php';
 
+$wgAvailableRights[] = 'commentcreate';
 $wgAvailableRights[] = 'commentmove';
 $wgAvailableRights[] = 'commentedit';
 $wgAvailableRights[] = 'commentdelete';
 
+$wgGroupPermissions['*']['commentcreate'] = true;
```

Comment:
It can go here if that's the expected default value. We call `WikiFactoryLoader::LocalToGlobalPermissions( $wgGroupPermissionsLocal );` at the end of LocalSettings (`$wgGroupPermissionsLocal` is the WikiFactory variable for local permission changes per wikia), so this can still be overridden in WikiFactory.